### DISCUSSION TITLE
SingleR for methylation

### DISCUSSION THREAD

**Discussion by LeaLe88** [LTLA/SingleR#261] [General]
Dear all, 

Just wondering whether SingleR can also be used for annotation of methylation cell types?
Or if not, if you have any recommendation for any package.

Thank you!

**Comment by j-andrews7** [comment]
Snagging the beta values and using that should be easy enough in theory. Seems like the sparsity might be a killer though.

Then again, imputation in scWGBS seems to not completely suck, so may be able to overcome it to a degree. Dunno, think this would take some experimenting. 

Label transfer/projection mapping would probably work fine. And random forest classifiers are pretty robust with methylation data and used for array classifications all the time, e.g. [the MNP classifier](https://www.molecularneuropathology.org/mnp/).

I'm guessing Aaron will have deeper insights.

**Reply by LTLA** [reply]
I don't work much with methylation data, but from what I remember, there is the question of what to do about CpG sites that don't have any coverage, i.e., the beta value is missing. I don't think **SingleR** knows what to do with missing values in the input data; technically, the correct approach is to just ignore such values, but that would entail some significant changes to **SingleR**'s internals. I suppose you could just put some "neutral" placeholder value, e.g., 0.5.

Putting that aside, it's probably best to use the M-value as input instead of the beta value. This is a bit more like expression data (not bounded to [0, 1]) and should make the marker detection steps a bit happier. In this case, the placeholder would be zero.
